bindings/gumjs/gumdukinterceptor.c

```diff
@@ -466,7 +502,10 @@ _gum_duk_interceptor_on_enter (GumDukInterceptor * self,
     }
     else
     {
-      gumjs_invocation_context_release (ctx, jic);
+      if (self->cached_invocation_context == jic)
```

Comment:
`if (self->cached_invocation_context == jic)`
should be
`if (jic == self->cached_invocation_context)`
for consistency with the on_leave (which I'd say reads better).